catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/controller/ScriptController.kt

```diff
@@ -0,0 +1,275 @@
+            when {
+                brick is SetLookBrick && brick.look != null ->
+                    brick.look =
+                        lookController.findOrCopy(brick.look, destinationScene, destinationSprite)
+
+                brick is WhenBackgroundChangesBrick && brick.look != null ->
+                    brick.look =
+                        lookController.findOrCopy(brick.look, destinationScene, destinationSprite)
+
+                brick is PlaySoundBrick && brick.sound != null ->
+                    brick.sound =
+                        soundController.findOrCopy(brick.sound, destinationScene, destinationSprite)
+
+                brick is PlaySoundAndWaitBrick && brick.sound != null ->
+                    brick.sound =
+                        soundController.findOrCopy(brick.sound, destinationScene, destinationSprite)
+
+                brick is UserVariableBrickInterface && brick.userVariable != null ->
+                    updateUserVariable(brick, destinationProject, destinationSprite)
+
+                brick is UserListBrick && brick.userList != null ->
+                    updateUserList(brick, destinationProject, destinationSprite)
+
+                brick is UserDataBrick ->
+                    updateUserData(brick, destinationProject, destinationSprite)
+            }
+        }
+        return script
+    }
+
+    @Suppress("NestedBlockDepth")
+    @Throws(CloneNotSupportedException::class)
+    fun pack(groupName: String?, bricksToPack: List<Brick>?) {
+        val scriptsToPack: MutableList<Script> = ArrayList()
+        val userDefinedBrickListToPack: MutableList<UserDefinedBrick> = ArrayList()
+        bricksToPack?.let {
+            for (brick in bricksToPack) {
+                if (brick is ScriptBrick) {
+                    if (brick is UserDefinedReceiverBrick) {
+                        val userDefinedBrick = brick.userDefinedBrick
+                        userDefinedBrickListToPack.add(userDefinedBrick.clone() as UserDefinedBrick)
+                    }
+                    val scriptToPack = brick.getScript()
+                    scriptsToPack.add(scriptToPack.clone())
+                }
+            }
+        }
+        BackpackListManager.getInstance()
+            .addUserDefinedBrickToBackPack(groupName, userDefinedBrickListToPack)
+        BackpackListManager.getInstance().addScriptToBackPack(groupName, scriptsToPack)
+        BackpackListManager.getInstance().saveBackpack()
+    }
+
+    @Throws(IOException::class, CloneNotSupportedException::class)
+    fun packForSprite(scriptToPack: Script, destinationSprite: Sprite) {
+        val script = scriptToPack.clone()
+        val scriptFlatBrickList: List<Brick> = ArrayList()
+        script.addToFlatList(scriptFlatBrickList)
+
+        for (brick in scriptFlatBrickList) {
+            when {
+                brick is SetLookBrick && brick.look != null ->
+                    brick.look = lookController.packForSprite(brick.look, destinationSprite)
```

Comment:
```suggestion
            when(brick) {
                is SetLookBrick, is WhenBackgroundChangesBrick ->
                    brick.look = brick.look?.let{lookController.packForSprite(it, destinationSprite)}
```
https://kotlinlang.org/docs/control-flow.html#when-expression
Improve all others too, please.